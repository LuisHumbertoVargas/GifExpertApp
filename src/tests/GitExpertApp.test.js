import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Prueba unitara del componente <GifExpertApp />', () => {
    test('Debe tomar una copia del componente <GifExpertApp />', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['Elon Musk', 'SpaceX'];
        const wrapper = shallow(
            <GifExpertApp defaultCategories={categories} />
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
});
