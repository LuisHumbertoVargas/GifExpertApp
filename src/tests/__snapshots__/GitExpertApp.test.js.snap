// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba unitara del componente <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Elon Musk"
      key="Elon Musk"
    />
    <GifGrid
      category="SpaceX"
      key="SpaceX"
    />
  </ol>
</Fragment>
`;

exports[`Prueba unitara del componente <GifExpertApp /> Debe tomar una copia del componente <GifExpertApp /> 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
